### PR TITLE
UX Improvments

### PR DESCRIPTION
**Pull request by justinkchen**

### CODE REVIEW

File: client/src/components/ProofGen/MailTable.tsx [hunk 1]
```diff
@@ -253,6 +276,11 @@ const ErrorContainer = styled.div`
   gap: 36px;
 `;
 
+const Icon = styled(SVGIconThemed)`
```
Comment (asoong) [inline client/src/components/ProofGen/MailTable.tsx:279]: I've been using `react-feather` for these. There should already be a mail icon.

File: client/src/components/RegistrationForm/NewRegistration.tsx [hunk 2]
```diff
@@ -36,10 +36,11 @@ export const NewRegistration: React.FC<NewRegistrationProps> = ({
 
   // ----- transaction state -----
   const [proof, setProof] = useState<string>('');
+  const [tx, setTx] = useState<string | null>(null);
```
Comment (asoong) [inline client/src/components/RegistrationForm/NewRegistration.tsx:39]: Prefer: `[submitRegistrationTransactionHash, setSubmitRegistrationTransactionHash]` or better

File: client/src/components/RegistrationForm/NewRegistration.tsx [hunk 3]
```diff
@@ -61,8 +62,8 @@ export const NewRegistration: React.FC<NewRegistrationProps> = ({
   const {
     config: writeSubmitRegistrationConfig
   } = usePrepareContractWrite({
-    address: rampAddress,
-    abi: rampAbi,
+    address: rampAddress ?? undefined,
+    abi: rampAbi ?? undefined,
```
Comment (asoong) [inline client/src/components/RegistrationForm/NewRegistration.tsx:66]: Need a better solution for the typing mismatch here. I've left all sc writes like this for now until we can address them all

File: client/src/components/Swap/OnRamp.tsx [hunk 4]
```diff
@@ -51,13 +51,14 @@ export const OnRamp: React.FC<OnRampProps> = ({
 
   const [shouldConfigureRampWrite, setShouldConfigureRampWrite] = useState<boolean>(false);
   const [shouldFetchVerifyProof, setShouldFetchVerifyProof] = useState<boolean>(false);
+  const [tx, setTx] = useState<string | null>(null);
```
Comment (asoong) [inline client/src/components/Swap/OnRamp.tsx:54]: Same, see above

File: client/src/components/Swap/OnRamp.tsx [hunk 5]
```diff
@@ -70,8 +71,8 @@ export const OnRamp: React.FC<OnRampProps> = ({
   const {
     data: verifyProofRaw,
   } = useContractRead({
-    address: sendProcessorAddress,
-    abi: sendProcessorAbi,
+    address: sendProcessorAddress ?? undefined,
+    abi: sendProcessorAbi ?? undefined,
```
Comment (asoong) [inline client/src/components/Swap/OnRamp.tsx:75]: Is there a potential for a runtime error? If not I'd rather leave these until we address them all

File: client/src/components/modals/Modal.tsx [hunk 6]
```diff
@@ -295,6 +319,17 @@ export const Modal: React.FC<ModalProps> = ({
         >
           {ctaButtonTitle}
         </Button>
+
+        {transactionAddress?.length ? (
+          <Link
+            href={`https://goerli.etherscan.io/tx/${transactionAddress}`}
```
Comment (asoong) [inline client/src/components/modals/Modal.tsx:325]: Can you add the block scan link to `SmartContractContext` and pull that in here?

File: client/src/icons/svg/dark-mailbox.svg
Comment (asoong) [inline client/src/icons/svg/dark-mailbox.svg:1]: Shouldn't need this!

**Review by asoong — CHANGES_REQUESTED** [review]
Dooooooope:
- Can you post screens, I want to see the block explorer link
- Would suggest that instead of waiting to set the transaction on success, you set it on submit so they can see it while its mining
- Put the confetti behind a feature flag

**Comment by justinkchen** [comment]
Also moved confetti on submit of "Complete Order" as sachin mentioned, they are expecting it to be "done" and may navigate away by that point just like you said, they want to see the tx on etherscan so we will show confetti right away as it is in the process of completing at that point already.

**Comment by justinkchen** [comment]
https://www.dropbox.com/s/mv0vatm59my3nb4/Screen%20Recording%202023-11-13%20at%201.57.49%E2%80%AFAM.mov?dl=0

**Comment by justinkchen** [comment]
> Dooooooope:
> 
> * Can you post screens, I want to see the block explorer link
> * Would suggest that instead of waiting to set the transaction on success, you set it on submit so they can see it while its mining
> * Put the confetti behind a feature flag


Screen recording:

https://www.dropbox.com/s/mv0vatm59my3nb4/Screen%20Recording%202023-11-13%20at%201.57.49%E2%80%AFAM.mov?dl=0

**Comment by asoong** [comment]
LGTM

**Review by asoong — APPROVED** [review]

**Pull request merged**